clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/AbstractVeniceFilter.java

```diff
@@ -1,61 +1,31 @@
 package com.linkedin.venice.hadoop;
 
 import com.linkedin.venice.utils.VeniceProperties;
+import java.io.Closeable;
+import java.io.IOException;
 
 
 /**
  * An abstraction to filter data using Chain of Responsibility pattern.
  * @param <INPUT_VALUE>
  */
-public abstract class AbstractVeniceFilter<INPUT_VALUE> {
+public abstract class AbstractVeniceFilter<INPUT_VALUE> implements Closeable {
   protected final VeniceProperties props;
 
   public AbstractVeniceFilter(final VeniceProperties props) {
     this.props = props;
   }
 
-  private AbstractVeniceFilter<INPUT_VALUE> next;
-
   /**
    * This function implements how to parse the value and determine if filtering is needed.
    * @param value
    * @return true if the value should be filtered out, otherwise false.
    */
-  protected abstract boolean apply(final INPUT_VALUE value);
-
-  /**
-   * This function passes the value through the filter chain to determine
-   * if the value can be filtered out or not.
-   * @param value
-   * @return true if the value should be filtered out by this filter or its successor, otherwise false.
-   */
-  public boolean applyRecursively(final INPUT_VALUE value) {
-    if (apply((value))) {
-      return true;
-    } else {
-      return next != null && next.applyRecursively(value);
-    }
-  }
+  public abstract boolean apply(final INPUT_VALUE value);
 
   /**
-   * Set the next filter for this chain.
-   * @param filter, a chainable filter to handle <INPUT_VALUE>.
+   * Close any resources that this filter holds.
+   * @throws IOException
    */
-  public void setNext(final AbstractVeniceFilter<INPUT_VALUE> filter) {
-    this.next = filter;
-  }
-
-  /**
-   * @return the next filter in this chain.
-   */
-  public AbstractVeniceFilter<INPUT_VALUE> next() {
-    return this.next;
-  }
-
-  /**
-   * @return true if the chain has more filters.
-   */
-  public boolean hasNext() {
-    return this.next != null;
-  }
+  public abstract void close();
```

Comment:
`Closeable` already defines a `close` method, why do we need to create a new one here?

Reply:
Good catch! I initially didn't have `Closeable` and forgot to delete this when added it!